testApi/src/main/scala/africa/absa/testing/scapi/rest/response/Response.scala

```diff
@@ -59,12 +59,13 @@ object Response {
    * The appropriate group's performAssertions method is called based on the group type of each Assertion.
    * Returns true only if all assertions return true, and false as soon as any assertion returns false.
    *
+   * @param environment The current environment.
    * @param response   The response on which actions will be performed.
-   * @param responseAction The set of response actions that dictate what actions will be performed on the response.
+   * @param responseActions The set of response actions that dictate what actions will be performed on the response.
    * @return           Boolean indicating whether all response actions passed (true) or any response action failed (false). TODO
    * @throws IllegalArgumentException If an response action group is not supported.
    */
-  def perform(response: Response, responseAction: Seq[ResponseAction]): Try[Unit] = {
+  def perform(environment: Environment, response: Response, responseActions: Seq[ResponseAction]): Try[Unit] = {
     def logParameters(response: Response, resolvedResponseAction: ResponseAction, exception: Option[Throwable] = None): Unit = {
       val filteredParams = resolvedResponseAction.params.filter(_._1 != "method").map { case (k, v) => s"$k->$v" }.mkString(", ")
```

Comment:
For readability, I would probably change this to
```suggestion
      val filteredParams = resolvedResponseAction.params
        .filter { case (k, _) => k != "method"}
        .map { case (k, v) => s"$k->$v" }
        .mkString(", ")
```
.

The possible alternative of 
```suggestion
      val filteredParams = resolvedResponseAction.params
        .flatMap {
          case (k, v) if k != "method" => Some(s"$k->$v")
          case _ => None
        }
        .mkString(", ")
```
may be too convoluted.

Reply:
Used more functional solution with flatmap. See in commit a446cfff190bbb995b33cfc8b677cfba6bff48cd.